.gitignore

```diff
@@ -73,8 +73,8 @@ public
 
 # Editor/IDE
 .idea
-.vscode
-!.vscode/settings.json
+-.vscode
```

Comment:
I think / guess you actually want to achieve this:  (.vscode folder not tracked except the settings file)
```suggestion
.vscode/*
.vscode/settings.json
```

git doesn't even list subfolders so unless you have a more specific exclude using a wildcard you can't re-include content in a previously excluded folder.

Reply:
Hmm are you sure? This works fine for me

<img width="796" alt="image" src="https://user-images.githubusercontent.com/1110551/68934585-393d5b80-0797-11ea-8d25-ecb9035cb32a.png">

Reply:
These don't work for me:
```
-.vscode
-!.vscode/settings.json
```

Reply:
@davifantasia  this branch is already merged, please open an issue if the current master does not behave as expected in your setup. Thanks !